src/tests/multihost/alltests/test_proxy_rfc2307.py

```diff
@@ -0,0 +1,630 @@
+                execute_cmd(multihost, command)
+        # Waiting for negative cache to expire - default 15 seconds
+        time.sleep(16)
+        for command in ["getent -s sss passwd newuser",
+                        "getent -s sss group newgroup"]:
+            execute_cmd(multihost, command)
+
+    def test_nested_group(self, multihost, backupsssdconf):
+        """
+        :title: nested group
+        :id: b35a6a2e-0ce5-11ed-b2dc-845cf3eff344
+        :steps:
+          1. Look for nested group/user
+        :expectedresults:
+          1. Should succeed
+        """
+        ldap_uri = 'ldap://%s' % multihost.master[0].sys_hostname
+        ds_rootdn = 'cn=Directory Manager'
+        ds_rootpw = 'Secret123'
+        ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+        for (cn, uid) in [('user1', '111112'), ('user2', '111113')]:
+            user_info = {'cn': cn.encode('utf-8'),
+                         'homeDirectory': f'/export/{cn}'.encode('utf-8'),
+                         'objectClass': [b'account',
+                                         b'posixAccount'],
+                         'uidNumber': uid.encode('utf-8'),
+                         'gidNumber': uid.encode('utf-8'),
+                         'uid': cn.encode('utf-8')}
+            user_dn = f'uid={cn},ou=Users,{ds_suffix}'
+            (_, _) = ldap_inst.add_entry(user_info, user_dn)
+
+        for (cn, uid, memberUid) in [('childgroup', '111113', 'user2'.encode('utf-8')),
+                                     ('middlegroup', '111114', 'childgroup'.encode('utf-8')),
+                                     ('topgroup', '111115', ['middlegroup'.encode('utf-8'),
+                                                             'user1'.encode('utf-8')])]:
+            grop_info = {'cn': cn.encode('utf-8'),
+                         'objectClass': b'posixGroup',
+                         'gidNumber': uid.encode('utf-8'),
+                         'memberUid': memberUid}
+            group_dn = f'cn={cn},ou=Groups,{ds_suffix}'
+            (_, _) = ldap_inst.add_entry(grop_info, group_dn)
+        tools = sssdTools(multihost.client[0])
+        sssd_params = {'domains': ds_instance_name}
+        tools.sssd_conf('sssd', sssd_params)
+        domain_name = tools.get_domain_section_name()
+        domain_params = {'debug_level': '9',
+                         'id_provider': 'proxy',
+                         'proxy_lib_name': 'ldap',
+                         'proxy_pam_target': 'sssdproxyldap'}
+        tools.sssd_conf('domain/' + domain_name, domain_params)
+        tools.clear_sssd_cache()
+        for command in ["id user1 | grep topgroup",
+                        "id user2 | grep childgroup",
+                        "getent group middlegroup",
+                        "getent group topgroup | grep user1",
+                        "getent group childgroup | grep user2"]:
+            execute_cmd(multihost, command)
```

Comment:
Clean up of users and groups not done

Reply:
cleaned up with class fixture : ldap_objects_sssd_client